### DISCUSSION TITLE
Discrepancy between output voltage in SPMe model and literature (Marquis 2019)

### DISCUSSION THREAD

**Discussion by SimMLPhys** [pybamm-team/PyBaMM#4584] [General]
In the 2019 Marquis paper, the following overall voltage equation is provided:
$$V = U_{\text{eq}} + \eta_r + \eta_c + \Delta\Phi_{\text{Elec}} + \Delta\Phi_{\text{Solid}}$$

However, when I try to check using a SPMe in python, I find this relation does not exactly hold:

```
U_eq = sim.solution['Battery open-circuit voltage [V]'].data
eta_r = sim.solution['X-averaged reaction overpotential [V]'].data
eta_c = sim.solution['X-averaged concentration overpotential [V]'].data
ohmic_electrolyte = sim.solution['X-averaged battery electrolyte ohmic losses [V]'].data
ohmic_solid = sim.solution['X-averaged solid phase ohmic losses [V]'].data
volt = sim.solution['Voltage [V]'].data
if np.array_equal(U_eq + eta_r + eta_c + ohmic_electrolyte + ohmic_solid, volt):
    print("The sum of arrays is equal to voltage.")
else:
    print("The sum of arrays is not equal to voltage.")
```

> The sum of arrays is not equal to voltage.

This is the model I am using:

```
def create_custom_experiment():    
    experiment = pybamm.Experiment([
        (f"Rest for 200 seconds", f"Charge at 0.2C for 200 seconds or until 4.2 V", f"Rest for 200 seconds", 
         f"Charge at 1C for 10 seconds or until 4.2 V", f"Rest for 200 seconds", 
         f"Discharge at 1C for 10 seconds or until 3.0 V")
    ])
    return experiment

default_options = {"thermal":"isothermal"}
model_SPMe_1 = pybamm.lithium_ion.SPMe(options=default_options)

custom_experiment = create_custom_experiment()
sim = pybamm.Simulation(model_SPMe_1,
                        experiment=custom_experiment)


sim.solve()

```
It looks like there is a slightly higher overpotential overall than from just the sum:

![image](https://github.com/user-attachments/assets/ba15be30-6b4b-40ec-86e7-904dd324035a)

I am hoping someone can help me find out where this discrepancy is coming from.

**Comment by martinjrobins** [comment]
Hi @SimMLPhys, the numerical solvers in PyBaMM will only solve the equations up to a certain tolerance. You are able to reduce the tolerance (i.e. make the solution more accurate) via the arguments to the solvers, see for example https://docs.pybamm.org/en/stable/source/api/solvers/idaklu_solver.html. Numerical errors in the solution can potentially build up over long-running experiments, and will be influenced by discontinuities between different experimental steps. If you are concerned about numerical errors in your solution, I would perform a convergence study, reducing the tolerances and evaluating how quickly your solution converges.

**Reply by SimMLPhys** [reply]
I see. So it's not that the overall voltage actually has an additional term added to it that was missing from the analytical expression, it's just a numerical issue. Thanks.

**Comment by rtimms** [comment]
Also see https://github.com/pybamm-team/PyBaMM/issues/4145
